main.go

```diff
@@ -62,6 +63,7 @@ func main() {
 	var (
 		app                 = kingpin.New(filepath.Base(os.Args[0]), "The Pushgateway")
 		webConfig           = webflag.AddFlags(app)
+		gzipLevel           = app.Flag("gziplevel", "gzip compress level").Default("-1").Int() // gzip.DefaultCompression
```

Comment:
The doc string should be formatted as a sentence (start with capital letter, end with period), and maybe be a bit more descriptive (what levels are allowed, what's their meaning). See the other doc strings for comparison. Also, it seems to be more common to me to call it "compression level", not "compress level". And final nit: the flag itself should follow the structure of the other flags, so I would suggest "web.gzip-level" rather than "gziplevel".

Reply:
In different news, the default behavior should be to not do compression, even if the client requests it (because that's the current behavior, and compression might actually make things worse as discussed in #84, so we cannot just change behavior without cutting a major version).

I _think_ you can just pass in compression level 0 for "no compression", but I'm not sure if the gzip handler then really switches into "pass through" mode without overhead. So perhaps make the default here 0, and if the gzip handler handles that without overhead, we are good. Otherwise, you could simply add a condition below to not wrap with the gzip handler if the gzipLevel is 0.